coderd/provisionerdserver/provisionerdserver_test.go

```diff
@@ -2189,7 +2189,7 @@ type overrides struct {
 	externalAuthConfigs         []*externalauth.Config
 	templateScheduleStore       *atomic.Pointer[schedule.TemplateScheduleStore]
 	userQuietHoursScheduleStore *atomic.Pointer[schedule.UserQuietHoursScheduleStore]
-	timeNowFn                   func() time.Time
+	clock                       quartz.Clock
```

Comment:
We can probably set this as a `*quartz.Mock` here since that's likely what we'll eventually be using.